server/src/test/java/org/apache/kafka/server/AssignmentsManagerTest.java

```diff
@@ -172,10 +172,11 @@ public void testAssignmentAggregation() throws InterruptedException {
         manager.onAssignment(new TopicIdPartition(TOPIC_1, 3), DIR_3, () -> { });
         manager.onAssignment(new TopicIdPartition(TOPIC_1, 4), DIR_1, () -> { });
         manager.onAssignment(new TopicIdPartition(TOPIC_2, 5), DIR_2, () -> { });
-        while (!readyToAssert.await(1, TimeUnit.MILLISECONDS)) {
+        TestUtils.waitForCondition(() -> {
             time.sleep(100);
```

Comment:
IIRC, `TestUtils.waitForConditio` includes the `sleep`, so you don't need to call sleep manually

Reply:
I tried to remove the sleep but the test would failed because the readyToAssert.await() would not work as expected. Any ideas on that? Thanks!

Reply:
oh, it is `MockTimer` so we call `sleep` to advance the time. please ignore my comment :)